metrics/mock_metrics/mock_scope.go

```diff
@@ -0,0 +1,145 @@
+// Automatically generated by MockGen. DO NOT EDIT!
```

Comment:
As an example of where this is useful, see the test cases I just added for the stats-per-log. I'm able to `EXPECT()` a `NewScope` call, and then leave off the statName part of subsequent logs.

Also, it would require a fair bit of moving around of test code to switch to using a `MockStatter` underneath, and we're likely to refactor this soon anyhow, with the move to Prometheus.